scona/visualisations.py

```diff
@@ -0,0 +1,314 @@
+import warnings
+
+import numpy as np
+import networkx as nx
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+
+def plot_rich_club(rc_coef, rc_coef_rand, figure_name=None, color=None,
+                   x_max=None, y_max=None):
+    """
+    This is a visualisation tool for plotting the rich club values per degree
+    along with the random rich club values created from a random network
+    with a preserved degree distribution.
+
+    Parameters
+    ----------
+    rc_coef : dict
+        rich club coefficient values
+    rc_coef_rand : :dict
+        random rich club coefficient values
+    figure_name : str, optional
+        path to the file to store the created figure in (e.g. "/home/Desktop/name")
+        or to store in the current directory include just a name ("fig_name");
+    color : list of 2 strings, optional
+        where the 1st string is a color for rich club values and 2nd - for random
+        rich club values. You can specify the color using an html hex string
+        (e.g. color =["#06209c","#c1b8b1"]) or you can pass an (r, g, b) tuple,
+        where each of r, g, b are in the range [0,1]. Finally, legal html names
+        for colors, like "red", "black" and so on are supported.
+    x_max : int, optional
+        the max length of the x-axis of the plot
+    y_max : int, optional
+        the max length of the y-axis of the plot
+
+    Returns
+    -------
+        Plot the Figure and if figure_name provided, save it in a figure_name file.
+
+    """
+
+    # get the degrees
+    degree = list(rc_coef.keys())
+
+    # get the rich club coefficient values
+    rc = list(rc_coef.values())
+
+    # get the random rich club coefficient values
+    rc_rand = list(rc_coef_rand.values())
```

Comment:
So to get these rich club values for your random graphs you would call `B.report_rich_club()` where `B` is your [GraphBundle (see docs)](https://whitakerlab.github.io/scona/scona.html?highlight=graphbundle#scona.classes.GraphBundle) containing all your random graphs.

I think ideally we should be passing a GraphBundle directly to the plotting function, since this is the scona class for handling inter network comparisons, unfortunately the documentation on this subject is a little sparse and I haven't had a chance to talk Ruslan through it yet

Reply:
@KirstieJane So we need to plot rich club coefficient values obtained not only from one random graph, but rich club coefficient values with error bars from a bunch of random graphs, right? 
(at this moment, we are plotting real rich club values and random rich club values from 1 random graph to compare with)

Reply:
@Islast @KirstieJane What do you think about passing a **parameter**  num_rand_measures=Integer` to the function and based on the value (0 or 3 or 10) we create the corresponding amount of random graphs and then random measures?
Like this:
```
for i in range(num_rand_measures):
    R = random_graph(H)
    
    # Calculate the rich club coefficient
    rc_rand_dict = rich_club(R)

    # And save the values to the numpy array
    rc_rand[:, i] = list(rc_rand_dict.values())
```
But in my previous implementation, I took the mean of network measures and passed *real_network_measures* and *random_network_measures* to the function to plot. 
But I should plot all the random network measures, right?

Reply:
Having an option to generate a bunch of random graphs on the fly is not a bad idea. 
The ideal way to generate new random graphs is to initiate a `GraphBundle` object.
```python
brain_bundle = scn.GraphBundle([H], ['trueNetwork'])
brain_bundle.create_random_graphs('trueNetwork', num_random_graphs)
```
You can get a dataframe of rich club values from this by calling `brain_bundle.report_rich_club()`.  When you calculate error bars/mean etc from this dataframe remember to mask out the original graph `'trueNetwork'`.

Reply:
@Islast 
I understood the scona way of dealing with across network comparisons by using GraphBundle.
I have redesigned the plotting functions to use `GraphBundle` object as a parameter. 
! But I have an assumption that the when we calculate the *rich_club_values* (`brain_bundle.report_rich_club()`) the *1st column will be our real network*, all others - random networks. Is this always true? (I followed the process of creating random graphs and reporting rich club, but still, I am not 100% sure that real_network will **always** be the 1st column while **reporting rich club**.

Otherwise, I also have to pass the *name of the Real Network* to plotting function.

Right now the signature is `plot_rich_club(brain_bundle,...)`. If I am mistaken with my a-little-tested assumption, the signature would be `plot_rich_club(brain_bundle, name_realGraph)``

Reply:
My personal preference in these matters is to be super explicit about everything, so I would pass the name of the real network
If you prefer you could have default set to first column with an option to pass a name - as long as the docstrings are **super clear** about what is going on

Reply:
Okay, I will make changes to accept *real_network_name* as a parameter to function

Reply:
I see now that it was I who introduced the term __real network__ the greatest monster is always in the mirror :moyai: